src/test/java/com/bbtutorials/users/dao/WorkoutDaoTests.java

```diff
@@ -0,0 +1,133 @@
+package com.bbtutorials.users.dao;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.local.embedded.DynamoDBEmbedded;
+import com.amazonaws.services.dynamodbv2.model.*;
+import com.bbtutorials.users.model.Workout;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.Assert.*;
+
+public class WorkoutDaoTests {
+    private static AmazonDynamoDB amazonDynamoDB;
+    private static DynamoDBMapper mapper;
+    private static WorkoutDao workoutDao;
+
+
+    private final String testEmail = "test@mail.com";
+    private final String testName = "Name";
+    private final String testId = "test";
+
+    @BeforeClass
+    public static void setup() {
+        System.setProperty("sqlite4java.library.path", "native-libs");
+
+        amazonDynamoDB = DynamoDBEmbedded.create().amazonDynamoDB();//dynamoDB.getAmazonDynamoDB();
+        mapper = new DynamoDBMapper(amazonDynamoDB);
+
+        amazonDynamoDB.createTable(new CreateTableRequest()
+                .withTableName("Workouts")
+                .withKeySchema(new KeySchemaElement().withAttributeName("Id").withKeyType(KeyType.HASH))
+                .withAttributeDefinitions(
+                        new AttributeDefinition().withAttributeName("Id").withAttributeType(ScalarAttributeType.S))
+                .withProvisionedThroughput(new ProvisionedThroughput(5L, 5L))
+        );
+
+        workoutDao = new WorkoutDao(mapper);
+    }
+
+    public Workout createWorkout() {
+        Workout testWorkout = new Workout();
+        testWorkout.setId(testEmail);
+        testWorkout.setUserEmail(testWeightName);
+        testWorkout.setName(testLbs);
+        
+        Set<String> testExer = new HashSet<String>();
+        Set<Integer> testDays = new HashSet<Integer>();
+        
+        testExer.add("testexcercise1");
+        testExer.add("testexcercise2");
+        testWorkout.setExercises(testExer);
+        
+        testDays.add(1);
+        testDays.add(2);
+        testWorkout.setDaysOfWeek(testDays);
+
+        return testWorkout;
+    }
+
+
+    @Test
+    public void testPutGetRemoveWorkout() {
+        workoutDao.putWorkout(createWorkout());
+
+        List<Workout> workouts = workoutDao.getWorkouts(testEmail);
+        
+        Set<Integer> checkDays = workouts.get(0).getDaysOfWeek();
+        Set<String> checkExer = workouts.get(0).getExcercises();
+        
+        assertEquals(1, workouts.size());
+        assertEquals(testName, workouts.get(0).getName());
+        assertEquals(testId, workouts.get(0).getId());
+        assertEquals(testEmail, workouts.get(0).getUserEmail());
+        assertEquals(2, checkDays.size());
```

Comment:
You should just check for equality of the hash sets rather than size to make sure the elements are the same inside.

Reply:
You can use assertEquals() on hash sets here to simplify it and make the test more resilient to changes